sonar-orchestrator/src/main/java/com/sonar/orchestrator/util/ZipUtils.java

```diff
@@ -87,4 +90,11 @@ static void nativeUnzip(File zip, File toDir) {
       throw new IllegalStateException("Fail to unzip " + zip + " to " + toDir);
     }
   }
+
+  private static void verifyInsideTargetDirectory(ZipEntry entry, Path entryPath, Path targetDirPath) {
+    if (!entryPath.normalize().startsWith(targetDirPath.normalize())) {
```

Comment:
@simonbrandhof how slow is normalize?

This method is going to be called for every file in the zip, which for a report can be a large number, therefor, if normalize is any costly, not calling `targetDirPath.normalize()` could bring perf benefit

Reply:
On my box difference is not significative (~50 ms on total of 2 seconds) when unzipping the SonarQube zip file, which is 156MB.

Moreover note that the Java implementation of ZipUtils is called only when the native command `unzip` is not present on the *nix box. In the context of SonarSource environment that means only Windows machines.

Reply:
> only Windows machines

which happen to usually be the most vulnerable and slowest machine  #abitoftrollingonmondaymorning

more seriously, if you have a SSD, your test is not significative.
on top of this, implementation of `targetDirPath.normalize()` is platform dependent, which implies that only a test on a Windows machine can start to give a rough estimate of the perf impact

shall we summon a poor fellow running windows as a his/her base OS to test this? #trollingwindowsissoeasy

Reply:
I don't consider that as a critical problem and won't try to fix it. Let's keep it in mind, or please provide a patch if you want :-)